### PR TITLE
Configuration values for automations

### PR DESCRIPTION
**Pull request by TheFes**
## Proposed change
Added configuration values for automation based on those already provided for scripts

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
- Link to parent pull request in the codebase: none
- Link to parent pull request in the Brands repository: none
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @TheFes, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#22178).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_docs/automation/yaml.markdown [hunk 1]
```diff
@@ -21,6 +21,89 @@ automation kitchen:
...
+  description: A description of the automation.
+  required: false
+  default: ''
+  type: string
+initial_state:
+  description: Used to define the state of your automation at startup.
+  required: false
+  type: boolean
+trace:
+  description:
```
Comment (frenck) [inline source/_docs/automation/yaml.markdown]: missing description

File: source/_docs/automation/yaml.markdown [hunk 2]
```diff
@@ -21,6 +21,89 @@ automation kitchen:
...
+  description: A unique id for your automation, will allow you to make changes to the name and entity_id in the GUI, and will enable debug traces.
+  required: false
+  type: string
+description:
+  description: A description of the automation.
+  required: false
+  default: ''
+  type: string
+initial_state:
+  description: Used to define the state of your automation at startup.
```
Comment (frenck) [inline source/_docs/automation/yaml.markdown]:
```suggestion
  description: Used to define the state of your automation at startup. When not set, the state will be restored from the last run.
```

File: source/_docs/automation/yaml.markdown [hunk 3]
```diff
@@ -21,6 +21,89 @@ automation kitchen:
...
+  type: boolean
+trace:
+  description: 
+  required: false
+  default: {}
+  type: map
+  keys:
+    stored_traces:
+      description: The number of traces which will be stored
+      type: int
```
Comment (frenck) [inline source/_docs/automation/yaml.markdown]:
```suggestion
      type: integer
```

**Review by frenck — COMMENTED** [review]
Two small tweaks, otherwise nice! 👍

**Comment by TheFes** [comment]
Sorry, pressed the wrong button regarding the missing description for trace. Will add that one

**Comment by TheFes** [comment]
Deploy seems to be failing on `Liquid Exception: Configuration key 'initial_state' is a boolean type and therefore, requires a default. in /opt/build/repo/source/_docs/automation/yaml.markdown`
But as also added by @frenck, there is no default to be added here, as it will use the current state in case `initial_state` is not provided. So how I'm not sure how to resolve this.

File: source/_docs/automation/yaml.markdown [hunk 4]
```diff
@@ -21,6 +21,89 @@ automation kitchen:
...
+  type: string
+description:
+  description: A description of the automation.
+  required: false
+  default: ''
+  type: string
+initial_state:
+  description: Used to define the state of your automation at startup. When not set, the state will be restored from the last run.
+  required: false
+  type: boolean
```
Comment (frenck) [inline source/_docs/automation/yaml.markdown:41]: It actually does have a default: True :)

```suggestion
  type: boolean
  default: Restored from last run
```

File: source/_docs/automation/yaml.markdown [hunk 5]
```diff
@@ -21,6 +21,89 @@ automation kitchen:
 
 You can add as many labeled `automation` blocks as you want.
 
+{% configuration %}
+alias:
+  description: Friendly name for the automation.
+  required: false
+  type: string
+id:
+  description: A unique id for your automation, will allow you to make changes to the name and entity_id in the GUI, and will enable debug traces.
```
Comment (frenck) [inline source/_docs/automation/yaml.markdown]:
```suggestion
  description: A unique id for your automation, will allow you to make changes to the name and entity_id in the UI, and will enable debug traces.
```

File: source/_docs/automation/yaml.markdown [hunk 6]
```diff
@@ -21,6 +21,89 @@ automation kitchen:
...
+  description: Variables that will be available inside your [templates triggers](automation/trigger/#template-trigger).
+  required: false
+  default: {}
+  type: map
+  keys:
+    PARAMETER_NAME:
+      description: The value of the variable. Any YAML is valid. Only [limited templates](/docs/configuration/templating/#limited-templates) can be used.
+      type: any
+mode:
+  description: "Controls what happens when script is invoked while it is still running from one or more previous invocations. See [Automation Modes](#automation-modes)."
```
Comment (frenck) [inline source/_docs/automation/yaml.markdown]:
```suggestion
  description: "Controls what happens when the script is invoked while it is still running from one or more previous invocations. See [Automation Modes](#automation-modes)."
```

**Comment by TheFes** [comment]
Hooray, all checkes passed!

**Review by frenck — APPROVED** [review]
Thanks, @TheFes 🎉

**Pull request merged**